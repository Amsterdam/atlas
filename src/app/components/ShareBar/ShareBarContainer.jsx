import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import ShareBar from './ShareBar';
import { hasPrintMode, showPrintMode, sharePage } from '../../../shared/ducks/ui/ui';
import getShareUrl from '../../../shared/services/share-url/share-url';

export const handlePageShare = (target, fn) => {
  fn(target);
  const redirectUrl = getShareUrl(target, window);

  if (redirectUrl) {
    window.open(redirectUrl, '_blank');
  }
};

export const handlePrintMode = (openPrintMode) => {
  openPrintMode();
};

const mapStateToProps = (state) => ({
  hasPrintButton: hasPrintMode(state)
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  sharePage,
  showPrintMode,
  handlePageShare,
  handlePrintMode
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(ShareBar);
