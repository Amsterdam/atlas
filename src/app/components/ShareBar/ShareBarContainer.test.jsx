import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import ShareBarContainer from './ShareBarContainer';

const initialState = {
  ui: {
    viewMode: 'print'
  }
};

describe('ShareBarContainer', () => {
  it('should render', () => {
    const store = configureMockStore()({ ...initialState });
    const component = shallow(<ShareBarContainer />, { context: { store } });
    expect(component).toMatchSnapshot();
  });
});
