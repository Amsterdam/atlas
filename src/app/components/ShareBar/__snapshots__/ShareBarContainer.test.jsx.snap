// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareBarContainer should render 1`] = `
<ShareBarComponent
  handlePageShare={[Function]}
  handlePrintMode={[Function]}
  hasPrintButton={false}
  sharePage={[Function]}
  showPrintMode={[Function]}
/>
`;
